fix(core): error when invoking callbacks registered via ViewRef.onDestroy

crisbeto opened the pull request:
Invoking a callback registered through `ViewRef.onDestroy` throws an error, because we weren't registering it correctly in the internal data structure. These changes also remove the `storeCleanupFn` function, because it was mostly identical to `storeCleanupWithContext` and was only used in one place.

Fixes #36213.

atscott commented:
[presubmit](https://test.corp.google.com/ui#id=OCL:316168790:BASE:316168821:1591994480151:179865b3)

crisbeto committed:
fix(core): error when invoking callbacks registered via ViewRef.onDestroy

Invoking a callback registered through `ViewRef.onDestroy` throws an error, because we weren't registering it correctly in the internal data structure. These changes also remove the `storeCleanupFn` function, because it was mostly identical to `storeCleanupWithContext` and was only used in one place.

Fixes #36213.

AndrewKushnir commented:
[Presubmit #2](http://test/OCL:318511608:BASE:318511639:1593196536710:93a7ecff)
